File: code_samples/r/data_structures_hash_table.r
# Data Structures: Hash Table
# AI/ML Training Sample

HashTable <- setRefClass(
    "HashTable",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- HashTable$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/typescript/data_structures_hash_table.ts
/**
 * Data Structures: Hash Table
 * AI/ML Training Sample
 */

interface IHashTable {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class HashTable implements IHashTable {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new HashTable();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { HashTable, IHashTable };
